serving-core/src/main/scala/com/stratio/sparkta/serving/core/actor/FragmentActor.scala

```diff
@@ -154,6 +168,20 @@ class FragmentActor(curatorFramework: CuratorFramework)
       }
     }
   }
+
+  private def listByPath(fragmentLocation: String): Option[List[FragmentElementModel]] = {
+    Try {
+      if (CuratorFactoryHolder.existsPath(fragmentLocation)) {
+        val children = curatorFramework.getChildren.forPath(fragmentLocation)
+        JavaConversions.asScalaBuffer(children).toList.map(element =>
+          read[FragmentElementModel](new String(curatorFramework.getData.forPath(s"$fragmentLocation/$element"))))
+      }
+      else List.empty[FragmentElementModel]
+    } match {
+      case Success(list) => Some(list)
+      case Failure(e) => None
+    }
+  }
 }
```

Comment:
Maybe it's a good idea change the signature of the method listByPath.
Also, the method getExsitingFragment could be more readable.
I've thought something like this:
```scala
def getExistingFragment(fragment: FragmentElementModel): Option[FragmentElementModel] =
  listByPath(fragmentPath(fragment.fragmentType))
    .dropWhile(!fragment.equals)
    .headOption
    .orElse {
      throw new ServingCoreException(ErrorModel.toString(
       new ErrorModel(ErrorModel.CodeExistsFragmentWithName,
         s"Fragment of type ${fragment.fragmentType} with name ${fragment.name} already exists."))
      )
    }

private def listByPath(fragmentLocation: String): List[FragmentElementModel] =
  Try {
    if (CuratorFactoryHolder.existsPath(fragmentLocation)) {
      val children = curatorFramework.getChildren.forPath(fragmentLocation)
      JavaConversions.asScalaBuffer(children).toList.map(element =>
        read[FragmentElementModel](new String(curatorFramework.getData.forPath(s"$fragmentLocation/$element"))))
    }
    else List.empty[FragmentElementModel]
  }.getOrElse(List.empty[FragmentElementModel])
```